stl/src/filesystem.cpp

```diff
@@ -748,6 +748,83 @@ __std_win_error __stdcall __std_fs_get_file_id(__std_fs_file_id* const _Id, cons
     return __std_win_error::_Success;
 }
 
+[[nodiscard]] __std_win_error __stdcall __std_fs_space(const wchar_t* const _Target, uintmax_t* const _Available,
+    uintmax_t* const _Total_bytes, uintmax_t* const _Free_bytes) noexcept {
+    // get capacity information for the volume on which the file _Target resides
+    static_assert(sizeof(uintmax_t) == sizeof(ULARGE_INTEGER) && alignof(uintmax_t) == alignof(ULARGE_INTEGER),
+        "Size and alignment must match for reinterpret_cast<PULARGE_INTEGER>");
+    const auto _Available_c   = reinterpret_cast<PULARGE_INTEGER>(_Available);
+    const auto _Total_bytes_c = reinterpret_cast<PULARGE_INTEGER>(_Total_bytes);
+    const auto _Free_bytes_c  = reinterpret_cast<PULARGE_INTEGER>(_Free_bytes);
+    if (GetDiskFreeSpaceExW(_Target, _Available_c, _Total_bytes_c, _Free_bytes_c)) {
+        return __std_win_error::_Success;
+    }
+
+    __std_win_error _Last_error{GetLastError()};
+    _Available_c->QuadPart   = ~0ull;
+    _Total_bytes_c->QuadPart = ~0ull;
+    _Free_bytes_c->QuadPart  = ~0ull;
+    if (_Last_error != __std_win_error::_Directory_name_is_invalid) {
+        return _Last_error;
+    }
+
+    // Input could have been a file; canonicalize and remove the last component.
+    // We use VOLUME_NAME_NT because it always has a mapping available and we don't care about the canonical path
+    // being "ugly" due to needing the _Dos_to_nt_prefix.
+    constexpr wchar_t _Dos_to_nt_prefix[]    = LR"(\\?\GLOBALROOT)";

```

Comment:
This should be a `static constexpr` array. The difference is significant. An automatic `constexpr` array will emit codegen to push every (compile-time constant) value onto the stack, because arrays are address-taken at the point of use, and different automatic arrays need distinct addresses. A `static constexpr` array lives directly within read-only memory and the codegen just emits a pointer to it.